lib/go/edgecontext/service.go

```diff
@@ -10,14 +10,60 @@ const servicePrefix = "service/"
 // service talking to us.
 type Service AuthenticationToken
 
+func (s Service) isService() bool {
+	subject := AuthenticationToken(s).Subject()
+	return strings.HasPrefix(subject, servicePrefix)
+}
+
 // Name returns the name of the service.
 //
 // If it's not coming from an authenticated service,
 // ("", false) will be returned.
 func (s Service) Name() (name string, ok bool) {
-	subject := AuthenticationToken(s).Subject()
-	if strings.HasPrefix(subject, servicePrefix) {
+	if s.isService() {
+		subject := AuthenticationToken(s).Subject()
 		return subject[len(servicePrefix):], true
 	}
 	return
 }
+
+// OnBehalfOfID returns the ID of the user on whose behalf the service is acting.
+//
+// If it's not coming from an authenticated service,
+// ("", false) will be returned.
+func (s Service) OnBehalfOfID() (id string, ok bool) {
+	if s.isService() {
+		token := AuthenticationToken(s)
+		if token.OnBehalfOf == nil {
+			return
```

Comment:
the original go code was written long time ago that we have slightly different style guidance now, we prefer not to use naked returns and always return explicitly:
```suggestion
			return "", false
```
same for the rest

Reply:
makes sense! i'll update the new code to follow that style